SConstruct

```diff
@@ -17,6 +17,12 @@ DESIRED_FLAGS = ['-Wno-gnu-designator',
                  '-Wno-tautological-constant-out-of-range-compare',
                  '-Wframe-larger-than=4096']
 
+# Added for SDL Task CT158
+DESIRED_FLAGS.extend(['-fno-strict-overflow', '-fno-delete-null-pointer-checks', '-fwrapv'])
+
+# Added for SDL Task T217
+DESIRED_FLAGS.extend(['-fstack-protector-strong', '-fstack-clash-protection', '-D_FORTIFY_SOURCE=2'])
```

Comment:
(pylint-line-too-long) Line too long (101/80)

Reply:
I'm not familiar with these flags, do we want them in for production or just for testing?  There are some debugging compiler flags that are good for testing but have a performance penalty we can't accept for release.

This list of for compiler flags that will be tested for and used if (I think) both gcc and g++ support them, the -D option should be handled differently, probably by adding it to env.defines

Reply:
These are production binary hardening flags. They are given in the SDL task for release binaries. They are priority 6 tasks which means we are required to do them. I added them to desired flags because we also compile with clang and the Intel compiler and there is no guarantee that all of those compilers have these flags as well.